entry.js

```diff
@@ -10,9 +12,23 @@ function Entry(relativePath, basePath, mode, size, mtime) {
 }
 
 
-Entry.prototype.isDirectory = function() {
+Entry.prototype.isDirectory = function isDirectory() {
   /*jshint -W016 */
   return (this.mode & 61440) === 16384;
 };
 
+/**
+  On systems where we cannot symlink, do not add any special inequality cases
+  (ie always return true here, and only get `change` operations for files with
+  different `mtime`, `size` or `mode`).
+
+  On systems where we can symlink, also count a directory whose `linkDir` has
+  changed as different.  This allows us to handle the case where a directory
+  goes from being copied (existing in multiple inputPaths) to being symlinked
+  (existing in only one inputPath) or vice-versa.
+*/
+Entry.prototype.equals = function equals(otherEntry) {
+  return !canSymlink || (this.linkDir === otherEntry.linkDir);
```

Comment:
The annoying thing about the "subclass" approach is you wouldn't be able to use raw entry objects from `walkSync`